sherlock/internal/models/app_version.go

```diff
@@ -26,3 +29,60 @@ func (a AppVersion) GetCiIdentifier() CiIdentifier {
 		return CiIdentifier{ResourceType: "app-version", ResourceID: a.ID}
 	}
 }
+
+// GetAppVersionPathIDs iterates from one AppVersion to another, treating each one's parent like a linked list.
+// If a path is found in few enough iterations, it'll be returned according to these rules:
+// - It will always be exclusive of the end AppVersion
+// - It will be inclusive of the start AppVersion when possible
+// - It will be ordered by iteration (the last entry in the path is the one whose parent is the end AppVersion)
+func GetAppVersionPathIDs(db *gorm.DB, inclusiveStartID uint, exclusiveEndID uint) (path []uint, foundPath bool, err error) {
+	if inclusiveStartID == exclusiveEndID {
+		// If the start and end are the same, there's no path to calculate
+		return []uint{}, true, nil
+	}
+
+	type AppVersionPathEntry struct {
+		ID                 uint
+		ParentAppVersionID uint
+	}
+	var appVersionPath []AppVersionPathEntry
+	if err = db.Raw(
+		//language=SQL
```

Comment:
Explicitly tell GoLand or other JetBrains IDEs that this string is SQL (you'll get validation directly against Sherlock's database if you've connected to it from GoLand)